cluster/etcd_service_discovery.go

```diff
@@ -267,10 +274,15 @@ func (sd *etcdServiceDiscovery) getServerFromEtcd(serverType, serverID string) (
 
 // GetServersByType returns a slice with all the servers of a certain type
 func (sd *etcdServiceDiscovery) GetServersByType(serverType string) (map[string]*Server, error) {
-	if m, ok := sd.serverMapByType.Load(serverType); ok {
-		sm := m.(map[string]*Server)
-		if len(sm) > 0 {
-			return sm, nil
+	sd.mapByTypeLock.RLock()
+	defer sd.mapByTypeLock.RUnlock()
+	if m, ok := sd.serverMapByType[serverType]; ok {
```

Comment:
I don't like the nested if. Do you know if this works?

```
if m, ok := sd.serverMapByType[serverType]; ok && len(m) > 0 {
  ...
}
```

Reply:
Yeah, I agree this looks better, I'm going to update it